mapflow/functional/service/data_catalog.py

```diff
@@ -63,12 +63,62 @@ def __init__(self,
     # Mosaics CRUD
     def create_mosaic(self):
         dialog = CreateMosaicDialog(self.dlg)
-        dialog.accepted.connect(lambda: self.api.create_mosaic(dialog.mosaic(), callback=self.create_mosaic_callback))
+        dialog.accepted.connect(lambda: self.create_mosaic_from_options(dialog))
         dialog.setup()
         dialog.deleteLater()
-
+    
+    def create_mosaic_from_options(self, mosaic_dialog: CreateMosaicDialog):
+        mosaic = mosaic_dialog.mosaic()
+        if mosaic_dialog.createMosaicCombo.currentIndex() == 0: # empty mosaic
+            self.api.create_mosaic(mosaic,callback=self.create_mosaic_callback)
+        else:
+            if mosaic_dialog.createMosaicCombo.currentIndex() == 1: # mosaic from files
+                image_paths = QFileDialog.getOpenFileNames(QApplication.activeWindow(), self.tr("Choose image to upload"), 
+                                                           filter='TIF files (*.tif *.tiff)')[0]
+            else: # mosaic from layers
+                dialog = UploadRasterLayersDialog(self.dlg)
+                # Get layers paths on accepting layers dialog
+                image_paths = []
+                def get_paths():
+                    for item in dialog.listWidget.selectedItems():
+                        image_paths.append(item.data(Qt.UserRole))
+                dialog.accepted.connect(get_paths)
+                # Show all acceptable raster layers in dialog
+                layers = []
+                for layer in self.project.mapLayers().values():
+                    if Path(layer.source()).suffix.lower() in ['.tif', '.tiff']:
+                        layers.append(layer)
+                dialog.setup(layers)
+                dialog.deleteLater()
+            if image_paths:
+                self.api.create_mosaic_from_images(mosaic, 
+                                                   callback=self.create_mosaic_from_images_callback,
+                                                   callback_kwargs={'image_paths' : image_paths,
+                                                                    'mosaic_name' : mosaic.name},
+                                                   error_handler=self.create_mosaic_from_images_error_handler,
+                                                   error_handler_kwargs={'image_paths' : image_paths,
+                                                                         'mosaic_name' : mosaic.name},
+                                                   image_paths=image_paths)            
+    
     def create_mosaic_callback(self, response: QNetworkReply):
         self.get_mosaics()
+        self.view.enable_mosaic_images_preview()
+        self.dlg.mosaicTable.clearSelection()
+    
+    def create_mosaic_from_images_callback(self, response: QNetworkReply, image_paths: List, mosaic_name: str):
+        mosaic_id = json.loads(response.readAll().data())['mosaic_id']
+        if image_paths:
```

Comment:
What if we don't have any more images to upload? This will happen if we select only one image. 
We still need to notify the user about uploaded images, so maybe we should call the upload_images anyway?